src/main/java/com/example/lures/LureMapper.java

```diff
@@ -23,4 +23,8 @@ public interface LureMapper {
     @Insert("INSERT INTO lures (product, company, size, weight) VALUES (#{product}, #{company}, #{size}, #{weight})")
     @Options(useGeneratedKeys = true, keyProperty = "id")
     void insert(Lure lure);
+
+    @Select("SELECT * FROM lures WHERE product = #{product}")
+    List<Lure> diplicatedLure(String product);

```

Comment:
目的は重複を見つけるためのものですが、Mapperとしては「データベースに対してどんな操作を行うのか」に焦点をあてたメソッド名がいいかと思います。
```suggestion
    List<Lure> findByProduct(String product);

```